File: packages/health_connector_hc_android/android/src/main/kotlin/com/phamtunglam/health_connector_hc_android/mappers/health_measurement_unit_mappers/TimeDurationMapper.kt
package com.phamtunglam.health_connector_hc_android.mappers.health_measurement_unit_mappers

import com.phamtunglam.health_connector_hc_android.pigeon.TimeDurationDto
import kotlin.time.Duration
import kotlin.time.Duration.Companion.seconds
import kotlin.time.DurationUnit

/**
 * Converts a [TimeDurationDto] to a Kotlin [Duration] object.
 */
internal fun TimeDurationDto.toDuration(): Duration = seconds.seconds

/**
 * Converts a Kotlin [Duration] to a [TimeDurationDto].
 *
 * Uses seconds as the transfer unit for consistency.
 */
internal fun Duration.toDto(): TimeDurationDto = TimeDurationDto(
    seconds = toDouble(DurationUnit.SECONDS),
)

File: packages/health_connector_hk_ios/lib/src/mappers/measurement_unit_mappers/time_duration_mapper.dart
import 'package:health_connector_core/health_connector_core_internal.dart'
    show TimeDuration, sinceV1_0_0;
import 'package:health_connector_hk_ios/src/pigeon/health_connector_hk_ios_api.g.dart'
    show TimeDurationDto;
import 'package:meta/meta.dart' show internal;

/// Converts [TimeDuration] to [TimeDurationDto].
@sinceV1_0_0
@internal
/// Converts [TimeDuration] to [TimeDurationDto].
@sinceV1_0_0
@internal
extension TimeDurationToDto on TimeDuration {
  TimeDurationDto toDto() {
    // Uses seconds as the transfer unit for consistency.
    return TimeDurationDto(seconds: inSeconds);
  }
}

/// Converts [TimeDurationDto] to [TimeDuration].
@sinceV1_0_0
@internal
/// Converts [TimeDurationDto] to [TimeDuration].
@sinceV1_0_0
@internal
extension TimeDurationDtoToDomain on TimeDurationDto {
  TimeDuration toDomain() {
    return TimeDuration.seconds(seconds);
  }
}
